ASP/syntax_test_asp.asp

```diff
@@ -674,4 +674,15 @@ test = "hello%>
 '         ^ - meta.method.source.asp
  </body>
 '^^^^^^^ meta.tag.structure.any.html
+<p class="<% If True Then %>test<% End If %>"></p>
+'         ^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html punctuation.section.embedded.begin.asp
+'            ^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html keyword.control.flow.asp
+'                         ^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html meta.if.block.asp punctuation.section.embedded.end.inside-block.asp
+'                           ^^^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html meta.if.block.asp
+'                               ^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html meta.if.block.asp punctuation.section.embedded.begin.inside-block.asp
+'                                  ^^^^^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html keyword.control.flow.asp
+'                                         ^^ text.html.asp text.html.basic meta.tag.block.any.html meta.attribute-with-value.class.html string.quoted.double.html meta.class-name.html source.asp.embedded.html punctuation.section.embedded.end.asp - meta.if.block.asp
+<script type="text/javascript">
+<% If True Then %>var hello = "world";<% End If %>
```

Comment:
currently the scope inside the ASP `I`f block in the JavaScript is `text.html.asp text.html.basic source.js.embedded.html source.asp.embedded.html meta.if.block.asp text.html.basic`, which means it is not highlighted as JS, but as HTML currently. I would welcome suggestions on DRY ways of fixing this - I had the idea of changing the `html` context inside `HTML-ASP.sublime-syntax` to include a match for `<script` in the prototype, but then I think it would be necessary to duplicate all the contexts in `ASP.sublime-syntax` to push the JavaScript scope instead of HTML - and we would have the same problem about remembering the JS block scope here too...